flocker_bb/ec2.py

```diff
@@ -204,24 +191,339 @@ def get_size(driver, size_id):
+    )
+
+    @classmethod
+    def from_driver_parameters(
+            cls, region, identifier, secret_identifier, **kwargs):
+        """
+        Create an ``EC2`` libcloud ``NodeDriver`` using the supplied region and
+        credentials and supply that as the ``driver`` for ``EC2CloudDriver``
+        along with all the other keyword arguments it requires.
+
+        :param bytes region: An EC2 region slug.
+        :param bytes identifier: An EC2 API identifier.
+        :param bytes secret_identifier: An EC2 API "secret" identifier.
+        :param dict kwargs: Keyword arguments for ``EC2CloudDriver``.
+        :returns: An ``EC2CloudDriver``instance.
+        """
+        from libcloud.compute.providers import get_driver, Provider
+        ec2_driver_factory = get_driver(Provider.EC2)
+        driver = ec2_driver_factory(
+            key=identifier,
+            secret=secret_identifier,
+            region=region
+        )
+
+        return cls(driver=driver, region=region, **kwargs)
+
+    def log_failure_arguments(self):
+        return dict(
+            format=self._FAILED_MSG_FORMAT,
+            region=self.region,
+            zulip_subject="EC2 Instances"
+        )
+
+    def get_image(self):
+        log.msg(
+            format="Getting image for %(image_id)s; required tags %(tags)s",
+            image_id=self.image_id, tags=self.image_tags,
+        )
+        return get_newest_tagged_image(
+            # Only get images belonging to us.  There's a ton of public AMIs
+            # that are definitely irrelevant.
+            images=self.driver.list_images(ex_owner="self"),
+            name=self.image_id,
+            tags=self.image_tags,
+            get_tags=lambda image: image.extra['tags'],
+        )
+
+    def get_image_metadata(self):
+        image = self.get_image()
+
+        image_metadata = {
+            'image_id': image.id,
+            'image_name': image.name,
+        }
+        image_metadata.update(image.extra['tags'])
+        return image_metadata
+
+    def create(self):
+        """
+        Create and start a new EC2 instance.
+
+        All parameters are fixed to the values used to initialize this driver.
+        """
+        image = self.get_image()
+        return self.driver.create_node(
+            name=self.name,
+            size=get_size(self.driver, self.instance_type),
+            image=image,
+            ex_keyname=self.keypair_name,
+            ex_userdata=self.user_data,
+            ex_metadata=self.instance_tags,
+            ex_securitygroup=[self.security_name],
+        )
+
+
+@attributes(
+    ['driver', 'name', 'flavor', 'keypair_name', 'image_id', 'image_tags',
+     'instance_tags', 'user_data']
+)
+@implementer(ICloudDriver)
+class RackspaceCloudDriver(object):
+    """
+    A driver for creating ``Compute`` nodes on ``Rackspace``.
+    """
+    _FAILED_MSG_FORMAT = (
+        "Rackspace Instance [%(instance_id)s]"
+        # XXX: Fix this URL.
```

Comment:
The pattern seems to be:
 * `https://mycloud.rackspace.com/cloud/<tenant_id>/servers#compute%2CcloudServersOpenStack%2C<REGION_SLUG>/<instance_id>`